storage/statedb/database.go

```diff
@@ -468,6 +471,26 @@ func (db *Database) insertPruningMark(hash common.ExtHash, blockNum uint64) {
 		Number: blockNum,
 		Hash:   hash,
 	})
+	if db.savingPruningMarksInBundle {
+		db.pruningMarksInBundle = append(db.pruningMarksInBundle, database.PruningMark{
+			Number: blockNum,
+			Hash:   hash,
+		})
+	}
+}
+
+func (db *Database) SavePruningMarksInBundle() {
+	db.savingPruningMarksInBundle = true
+}
+
+func (db *Database) RevertPruningMarksInBundle() {
+	db.diskDB.DeletePruningMarks(db.pruningMarksInBundle)
```

Comment:
Also delete from `db.pruningMarks` please. In case they are not yet written into diskDB.

Reply:
The pruningMark travels like `trie -(1)insertPruningMarks-> Database -(2)Commit or Cap-> diskDB`. 
(1) may happen frequently, but (2) doesn't happen too often. So when you call `RevertPruningMarksInBundle` in the worker, the (2) step might not finished yet. So there is nothing to DeletePruningMarks.